### PR TITLE
Feat tag specific authors

### PR DESCRIPTION
**Pull request by Kawacrepe**
Wdyt ?

Is that, how we want to manage our flagged authors? Does someone get another idea?

Do we want to keep 2 distinct function ? ( extractAndOptimizeUsers |  extractAllAuthorsFromLibrary)

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -2,5 +2,15 @@
 import { Maintainer } from "@npm/types";
 import { Scanner } from "@nodesecure/scanner";
 
-export function extractAndOptimizeUsers(maintainers: Maintainer): Maintainer[] | [];
-export function extractAllAuthorsFromLibrary(library: Scanner.Payload): Maintainer[] | [];
+export function extractAndOptimizeUsers(maintainers: Maintainer): authorsResponse
+export function extractAllAuthorsFromLibrary(library: Scanner.Payload): authorsResponse
```
Comment (fraxken) [inline index.d.ts:6]: Missing argument here no ?

File: src/index.js [hunk 2]
```diff
@@ -2,31 +2,69 @@
...
-  return useLevenshtein(authors);
+  const response = useLevenshtein(authors);
+
+  return addFlagsInResponse(response, flags);
+}
+
+function addFlagsInResponse(authors, flags = []) {
+  const flagsResponse = [];
+
+  if (Object.keys(flags).length === 0) {
```
Comment (fraxken) [inline src/index.js:46]: Why Object.keys is flags is an Array ?

File: src/index.js [hunk 3]
```diff
@@ -46,7 +84,7 @@ function splitAuthorNameEmail(author) {
   };
 }
 
-function formatResponse(author, maintainers, publishers) {
+function formatResponse({ author, maintainers, publishers, packageName = null }) {
```
Comment (fraxken) [inline src/index.js:87]: Maybe add an option argument and destruct it in the function.

File: src/levenshtein.js [hunk 4]
```diff
@@ -76,6 +76,9 @@ export function useLevenshtein(authors) {
       || isSimilar(author.name, currAuthor.name, true) < KMaxLevenshtein) {
         author.email = author.email.length > currAuthor.email.length ? author.email : currAuthor.email;
         author.name = author.name.length > currAuthor.name.length ? author.name : currAuthor.name;
+        if (packageName) {
```
Comment (fraxken) [inline src/levenshtein.js:79]: Avoid coercion here (not clear to me). I guess you mean:

```js
if (typeof packageName === "string") {
}
```

**Review by fraxken — CHANGES_REQUESTED** [review]

**Comment by Kawacrepe** [comment]
I updated this PR according to your review in "authors", discord channel.

The most notable changes are:
- we can pass an array of flaggedAuthors.
- `extractAndOptimizeUsers` is removed.
-  the new payload follows this schema.
```javascript
[
  {
    {
    name: "Blake Embrey",
    email: "hello@blakeembrey.com",
    flagged: true,
    packages: [
      {
        homepage: "https://github.com/blakeembrey/array-flatten",
        spec: "array-flatten",
        versions: "3.0.0",
        isPublishers: false
      },
      {
        homepage: "https://github.com/pillarjs/path-to-regexp#readme",
        spec: "path-to-regexp",
        versions: "6.2.0",
        isPublishers: true
      }
  }
]
```

File: README.md [hunk 5]
```diff
@@ -24,9 +24,32 @@ $ yarn add @nodesecure/authors
...
 ```js
-import { extractAndOptimizeUsers } from "@nodesecure/authors";
-
-const authors = extractAndOptimizeUsers(authors);
+import { extractAllAuthorsFromLibrary } from "@nodesecure/authors";
+
+const authors = extractAllAuthorsFromLibrary(library);
+
+// Expect authors to be following this schema
+// [
```
Comment (fraxken) [inline README.md:32]: Il y a pas un soucis sur l'exemple ?

Comment (Kawacrepe) [reply README.md:32]: Effectivement! les flags

**Comment by fraxken** [comment]
The index.d.ts file should be moved to `./src` i think.

**Review by fraxken — APPROVED** [review]
LGTM

**Pull request merged**